codec-http/src/test/java/io/netty/handler/codec/http/websocketx/extensions/compression/PerMessageDeflateDecoderTest.java

```diff
@@ -176,17 +179,75 @@ public void testMultiCompressedPayloadWithinFrame() {
 
         byte[] finalPayload1 = new byte[100];
         uncompressedFrame.content().readBytes(finalPayload1);
-        assertTrue(Arrays.equals(finalPayload1, payload1));
+        assertArrayEquals(finalPayload1, payload1);
         byte[] finalPayload2 = new byte[100];
         uncompressedFrame.content().readBytes(finalPayload2);
-        assertTrue(Arrays.equals(finalPayload2, payload2));
+        assertArrayEquals(finalPayload2, payload2);
         uncompressedFrame.release();
     }
 
+    @Test
+    public void testCompressionSkipForBinaryFrame() {
+        EmbeddedChannel encoderChannel = new EmbeddedChannel(
+                ZlibCodecFactory.newZlibEncoder(ZlibWrapper.NONE, 9, 15, 8));
+        EmbeddedChannel decoderChannel = new EmbeddedChannel(new PerMessageDeflateDecoder(false, ALWAYS_SKIP));
+
+        byte[] payload = new byte[300];
+        random.nextBytes(payload);
+
+        WebSocketFrame binaryFrame = new BinaryWebSocketFrame(Unpooled.copiedBuffer(payload));
+        encoderChannel.writeOutbound(binaryFrame.copy());
+        WebSocketFrame outboundFrame = encoderChannel.readOutbound();
+
+        assertEquals(0, outboundFrame.rsv());
+        assertArrayEquals(payload, ByteBufUtil.getBytes(outboundFrame.content()));
+        assertTrue(outboundFrame.release());
+
+        decoderChannel.writeInbound(binaryFrame);
+        WebSocketFrame inboundFrame = decoderChannel.readInbound();
+
+        assertEquals(0, inboundFrame.rsv());
+        assertArrayEquals(payload, ByteBufUtil.getBytes(inboundFrame.content()));
+        assertTrue(inboundFrame.release());
+
+        assertEquals(0, binaryFrame.refCnt());
+    }
+
+    @Test
+    public void testCompressionSkipForTextFrame() {
+        WebSocketExtensionFilter textFrameSkip = new WebSocketExtensionFilter() {
+            @Override
+            public boolean isMustSkipped(WebSocketFrame frame) {
+                return frame instanceof TextWebSocketFrame;
+            }
+        };
+        EmbeddedChannel encoderChannel = new EmbeddedChannel(
+                ZlibCodecFactory.newZlibEncoder(ZlibWrapper.NONE, 9, 15, 8));
+        EmbeddedChannel decoderChannel = new EmbeddedChannel(new PerMessageDeflateDecoder(false, textFrameSkip));
+
+        String payload = "not compressed payload";
+
+        WebSocketFrame textFrame = new TextWebSocketFrame(payload);
+        encoderChannel.writeOutbound(textFrame.copy());
+        WebSocketFrame outboundFrame = encoderChannel.readOutbound();
+
+        assertEquals(0, outboundFrame.rsv());
+        assertEquals(payload, outboundFrame.content().toString(UTF_8));
+        assertTrue(outboundFrame.release());
+
+        decoderChannel.writeInbound(textFrame);
+        WebSocketFrame inboundFrame = decoderChannel.readInbound();
+
+        assertEquals(0, inboundFrame.rsv());
+        assertEquals(payload, inboundFrame.content().toString(UTF_8));
+        assertTrue(inboundFrame.release());
+
+        assertEquals(0, textFrame.refCnt());
```

Comment:
call `channel.finish()` and assert return value

Reply:
Done.